collector/os.go

```diff
@@ -146,41 +157,105 @@ type Win32_OperatingSystem struct {
 	Version                 string
 }
 
-func (c *OSCollector) collect(ch chan<- prometheus.Metric) (*prometheus.Desc, error) {
-	var dst []Win32_OperatingSystem
-	q := queryAll(&dst)
-	if err := wmi.Query(q, &dst); err != nil {
+func (c *OSCollector) collect(ctx *ScrapeContext, ch chan<- prometheus.Metric) (*prometheus.Desc, error) {
+	nwgi, err := netapi32.GetWorkstationInfo()
+	if err != nil {
+		return nil, err
+	}
+
+	gmse, err := sysinfoapi.GlobalMemoryStatusEx()
+	if err != nil {
+		return nil, err
+	}
+
+	currentTime := time.Now()
+	timezoneName, _ := currentTime.Zone()
+
+	// Get total allocation of paging files across all disks.
+	memManKey, err := registry.OpenKey(registry.LOCAL_MACHINE, `SYSTEM\CurrentControlSet\Control\Session Manager\Memory Management`, registry.QUERY_VALUE)
+	if err != nil {
+		return nil, err
+	}
+	pagingFiles, _, err := memManKey.GetStringsValue("ExistingPageFiles")
+	if err != nil {
+		return nil, err
+	}
+
+	if err := memManKey.Close(); err != nil {
+		return nil, err
+	}
```

Comment:
This is a possible leak, since it isn't executed if `GetStringsValue` errors. You'll need to put this in a defer :( 
As an alternative, perhaps pull the openkey/read value/close into a helper?

Reply:
Good catch... I got a bit tied up trying to figure out we would handle an error in Close(), because if we defer it will be ignored. They do this in the documentation for the package however, and from reading the winapi stuff it seems it will only error if we try and close a key that's not open/already closed. So I think in our case we can safely ignore it and just defer. Something to be wary of though.

Reply:
Thanks @carlpett, I did find the named returns workaround in my travels but I was cautious because I thought we could end up in a situation where we e.g. fail to open a key, then fail to close it, and the close error overwrites the open failure which would lead to confusion. Directly logging the error seems like a great workaround, so I'll keep that in mind. Cheers!